src/Internal/Workflow/Process/Scope.php

```diff
@@ -245,8 +246,8 @@ public function cancel(\Throwable $reason = null): void
 
         foreach ($this->onCancel as $i => $handler) {
             $this->makeCurrent();
-            $handler($reason);
             unset($this->onCancel[$i]);
+            $handler($reason);
```

Comment:
why?

Reply:
I'm not sure if there's a way to call cancel inside cancel through this callback, but I prefer to deduplicate the function call without waiting for it to happen.